ml4gw/dataloading/hdf5_dataset.py

```diff
@@ -0,0 +1,175 @@
+import warnings
+from typing import Sequence, Union
+
+import h5py
+import numpy as np
+import torch
+
+from ml4gw.types import WaveformTensor
+
+
+class ContiguousHdf5Warning(Warning):
```

Comment:
Is this meant to be passed to `warnings.warn` ?

Reply:
No it's just creating a unique warning type so that downstream applications can catch it explicitly. But I wasn't using it in the warning below so good catch there, added it in now.